go/engine/puk_roll.go

```diff
@@ -152,6 +152,24 @@ func (e *PerUserKeyRoll) inner(ctx *Context) error {
 		len(pukBoxes), pukPrev != nil, gen)
 	libkb.AddPerUserKeyServerArg(payload, gen, pukBoxes, pukPrev)
 
+	ekLib := e.G().GetEKLib()
+	var myUserEKBox *keybase1.UserEkBoxed
+	var newUserEKMetadata *keybase1.UserEkMetadata
+	if ekLib != nil && ekLib.ShouldRun(ctx.NetContext) {
+		signingKey, err := pukSeed.DeriveSigningKey()
+		if err != nil {
+			return err
+		}
+		merkleRoot, err := e.G().GetMerkleClient().FetchRootFromServer(ctx.NetContext, libkb.EphemeralKeyMerkleFreshness)
+		sig, boxes, newMetadata, myBox, err := ekLib.PrepareNewUserEKForPUK(ctx.NetContext, *merkleRoot, signingKey)
```

Comment:
we should at least log this error, what was your reasoning for not aborting if it occurs?

Reply:
Reasoning: Shameful oversight.